src/lru.ts

```diff
@@ -1,4 +1,12 @@
-	class LRU {
+	class LRU<T = any> {
```

Comment:
This file is a gist of this PR. We adhere to TS rules and provide hint where TS might need help.

Note: For this pr, I am attempting to do minimal type definitions. This can be improvised to have even internal variables typed.

Reply:
this is the fundamental problem with typescript... `any` is what the following `key`s need to be because number or string is valid, but then people will complain... because there's no value at that point from the typescript pov. code completion already works with an IDE, so ... I don't see the point of these actions taken... it doesn't impact how it runs.